### PR TITLE
Master l10n be invoice bba improvement hsh

### PR DESCRIPTION
**Pull request by hsh-odoo**
Description of the issue/feature this PR addresses:

Task : https://www.odoo.com/web#id=32254&view_type=form&model=project.task&action=333&active_id=131&menu_id=4720
Pad : https://pad.odoo.com/p/r.17db2da2ae10011ca598d4f71928acc6

Current behavior before PR:

- Improve BBA Module

Desired behavior after PR is merged:
- Add BBA module in the accounting Settings and make it a general setting per company (not a setting per res.partner anymore)
- Make sure payment matching (invoice + payment) works correctly when the structured communication match

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/l10n_be_invoice_bba/__manifest__.py [hunk 1]
```diff
@@ -33,8 +33,11 @@
     'depends': ['account'],
     'data' : [
         'data/mail_template_data.xml',
+        'data/partner_data.xml',
         'views/res_partner_view.xml',
         'views/account_invoice_view.xml',        
         'views/report_invoice.xml',
+        'views/res_config_view.xml',
     ],
+    'auto_install': True,
```
[the remark this answers is not in the document]
Comment (qdp-odoo) [reply addons/l10n_be_invoice_bba/__manifest__.py]: ok as discussed with Laura:
* l10n_be depends on this module
* this module is not auto_install = True

File: addons/account/models/account_move.py [hunk 2]
```diff
@@ -688,7 +688,8 @@ def get_reconciliation_proposition(self, account_id, partner_id=False):
             query = """
                     SELECT a.id, b.id
                     FROM account_move_line a, account_move_line b
-                    WHERE a.amount_residual = -b.amount_residual
+                    WHERE a.ref = b.ref
+                    AND a.amount_residual = -b.amount_residual
```
Comment (oco-odoo) [inline addons/account/models/account_move.py]: I think what qdp intended we he told you "first try to match the reference (not the residual amount at all)" was that you should remove this line and make no check on amount_residual field in this query ;)

File: addons/l10n_be_invoice_bba/__manifest__.py [hunk 3]
```diff
@@ -26,14 +25,13 @@
         **R..R =** Customer Reference without non-numeric characters, **SSS =** Sequence Number, **DD =** Check Digits  
         
 The preferred type of Structured Communication and associated Algorithm can be
-specified on the Partner records. A 'random' Structured Communication will
-generated if no algorithm is specified on the Partner record. 
+specified on the Accounting Settings.
```
Comment (oco-odoo) [inline addons/l10n_be_invoice_bba/__manifest__.py]: "in the Accounting Settings"

**Comment by smetl** [comment]
specs changed since https://github.com/odoo/odoo/commit/b5bb5bd421ad5c643f72602f6a0f97a4a0eb8711

new PR: https://github.com/odoo/odoo/pull/26154

**Pull request closed without merging**